### PR TITLE
Always kill processes and remove direct reference to component governance task

### PR DESCRIPTION
**Pull request by natemcmaster**
Addresses https://github.com/aspnet/AspNetCore/issues/7777

Component governance is automatically configured for all builds now.

cc @Eilon  - targeting release/3.0-preview3 because this is affecting builds on all branches

### CODE REVIEW

**Comment by natemcmaster** [comment]
By the way, this is what I mean by "automatically configured". A post-build step is added automatically on all builds running in dnceng.

imagei

File: .azure/pipelines/ci.yml [hunk 1]
```diff
@@ -79,10 +79,6 @@ jobs:
       displayName: Run sign check
       condition: eq(variables['_SignType'], 'real')
 
-    # Detect OSS Components in use in the product. Only needs to run on one OS in the matrix.
-    - task: ms.vss-governance-buildtask.governance-build-task-component-detection.ComponentGovernanceComponentDetection@0
-      displayName: Detect components
-      condition: and(succeeded(), eq(variables['system.pullrequest.isfork'], false))
```
Comment (dougbu) [inline .azure/pipelines/ci.yml]: Nice to know about the new build step 😸

**Review by dougbu — APPROVED** [review]

**Review by Eilon — APPROVED** [review]

File: eng/scripts/KillProcesses.ps1 [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
+_kill testhost.exe
+_kill iisexpress.exe
+_kill iisexpresstray.exe
+_kill w3wp.exe
+_kill msbuild.exe
+_kill vbcscompiler.exe
+_kill git.exe
+_kill vctip.exe
+_kill chrome.exe
+_kill h2spec.exe
```
Comment (dougbu) [inline eng/scripts/KillProcesses.ps1]: What does this really change? The failure message `taskkill` spits out wasn't horrendous.

Comment (natemcmaster) [reply eng/scripts/KillProcesses.ps1]: The default view in AzDO shows all stderr messages (even though we are ignoring them). I'm trying to address this:

image

**Pull request merged**